### PR TITLE
 add fuzzy search for all menu

### PR DESCRIPTION
**Pull request by orthagh**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

Eerly version of the feature.
To discuss and todos:
- [x] tests
- [x] documentation: https://github.com/glpi-project/doc/pull/16
- [x] ~~do we need to permits shortcut configuration (currently: alt+ctrl+g) ?~~ no
- [x] ~~do we add an icon in UI to encourage discover of the feature ?~~ no
- [x] responsive
- [x] clean files in lib ?
- maybe missing entries

Preview:
fuzzyglpi

### CODE REVIEW

File: ajax/fuzzysearch.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+/** @file
+* @brief
+*/
+
+$AJAX_INCLUDE = 1;
```
Comment (trasher) [inline ajax/fuzzysearch.php]: I this relevant?

File: lib/fuzzy/.editorconfig [hunk 2]
```diff
@@ -0,0 +1,11 @@
+; This file is for unifying the coding style for different editors and IDEs.
```
Comment (trasher) [inline lib/fuzzy/.editorconfig]: This file should be dropped

File: lib/fuzzy/.gitignore [hunk 3]
```diff
@@ -0,0 +1,8 @@
+coverage.html
```
Comment (trasher) [inline lib/fuzzy/.gitignore]: This file should be dropped

File: lib/fuzzy/.jshintrc [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (trasher) [inline lib/fuzzy/.jshintrc]: This file should be dropped

File: lib/fuzzy/.npmignore [hunk 5]
```diff
@@ -0,0 +1,12 @@
+coverage.html
```
Comment (trasher) [inline lib/fuzzy/.npmignore]: This file should be dropped

File: lib/fuzzy/.travis.yml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+language: node_js
```
Comment (trasher) [inline lib/fuzzy/.travis.yml]: This file should be dropped

File: lib/jqueryplugins/jquery.hotkeys.js [hunk 7]
```diff
@@ -0,0 +1,204 @@
+/*jslint browser: true*/
```
Comment (trasher) [inline lib/jqueryplugins/jquery.hotkeys.js]: Is this one used from the lib? If not, please drop it

File: tests/units/Html.php [hunk 8]
```diff
@@ -630,4 +630,78 @@ public function testManageRefreshPage() {
       $message = \Html::manageRefreshPage(3, '$(\'#mydiv\').remove();');
       $this->string($message)->isIdenticalTo($expected);
    }
+
+   public function testGenerateMenuSession() {
+      //login to get session
+      $auth = new \Auth();
+      $this->boolean($auth->login(TU_USER, TU_PASS, true))->isTrue();
+
+      $this->array($_SESSION)
```
Comment (trasher) [inline tests/units/Html.php]: This one may fail running concurrent tests I guess :/

File: tests/units/Html.php [hunk 9]
```diff
@@ -630,4 +630,78 @@ public function testManageRefreshPage() {
...
+      //login to get session
+      $auth = new \Auth();
+      $this->boolean($auth->login(TU_USER, TU_PASS, true))->isTrue();
+
+      // init menu
+      \Html::generateMenuSession(true);
+
+      // test modal
+      $modal = \Html::FuzzySearch('getHtml');
+      $this->string($modal);
```
Comment (trasher) [inline tests/units/Html.php]: Maybe check if the string at least contains something expected

File: ajax/fuzzysearch.php (hunk 1 above)
Comment (orthagh) [inline ajax/fuzzysearch.php]: ```$AJAX_INCLUDE = 1``` skipsthe load of the plugin.
As the menu is loaded in $_SESSION, i think we don't need the plugin load

File: lib/jqueryplugins/jquery.hotkeys.js (hunk 7 above)
Comment (orthagh) [inline lib/jqueryplugins/jquery.hotkeys.js]: yes, used by my code:  https://github.com/glpi-project/glpi/pull/2266/files#diff-d2f7163a7210590a2f543d3ca47214e9R14

File: ajax/fuzzysearch.php (hunk 1 above)
Comment (trasher) [inline ajax/fuzzysearch.php]: ok

File: lib/jqueryplugins/jquery.hotkeys.js (hunk 7 above)
Comment (trasher) [inline lib/jqueryplugins/jquery.hotkeys.js]: ok

**Pull request merged**